sound/soc/sof/topology.c

```diff
@@ -586,6 +586,10 @@ static int sof_copy_tuples(struct snd_sof_dev *sdev, struct snd_soc_tplg_vendor_
 				if (*num_copied_tuples == tuples_size)
 					return 0;
 			}
+
+			/* stop when we've found the required token instances */
+			if (found == num_tokens * token_instance_num)
```

Comment:
@ranj063, would it be more intuitive if:
```c
if ((i +1) == token_instance_num)
```
iow, check the same type of counter instead of the aggregated information?

Reply:
our outright limit the outer loop to `min(le32_to_cpu(array->num_elems), token_instance_num)` ?

Reply:
@ujfalusi the limiting the outer loop is surely not more intuitive but maybe the first suggestion is OK but in either case the end result will be the same as what we have now right? So i'd like to leave it as is if you dont mind.